File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> intVar;

        for (int i = 0; i < nums.size(); i++) {
            intVar[nums[i]]++;
        }

        for (const auto& value : intVar) {
            if (value.second > 1) {
                return true;
            }
    }
    return false;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left , right = 0, len(nums)-1
        minVal = 9999999999999999
        while (left < right):
            mid = (left + right) // 2

            if nums[mid] > nums[right]:
                left = mid + 1
            elif nums[mid] < nums[right]:
                right = mid

        return nums[right]